### PR TITLE
Soft Event User Deletion For Admins

### PR DESCRIPTION
**Pull request by subhankar-panda**
Adds soft deletion for admins.

Funnily enough, a lot of this code is from 6 months ago but uses `setState` even though there are redux actions and reducers for the stuff in there?

### CODE REVIEW

**Comment by subhankar-panda** [comment]
Let's hold off on landing this for now, I have a couple of changes to make

**Comment by RedbackThomson** [comment]
We already have a plugin for soft delete called "mongoose-delete". It is as simple as calling model.delete(), rather than setting our own flags. It also timestamps it when it deletes it.

File: src/assets/js/components/User.js [hunk 1]
```diff
@@ -83,11 +97,36 @@ class User extends React.Component {
...
     return (
       <div>
-        <h3>User <small>{this.props.user._id}</small></h3>
+        <div className="row">
+          <div className="col-md-6">
+            <h3>User <small>{this.props.user._id}</small></h3>
+          </div>
+          <div className="col-md-6">
+            <div className="float-right">
+              {this.state.showConfirm ?
```
Comment (RedbackThomson) [inline src/assets/js/components/User.js]: Might want to move all of this into its own method, and then call it conditionally. Could be re-used elsewhere.

File: src/assets/js/data/Api.js [hunk 2]
```diff
@@ -197,6 +197,13 @@ export const deleteAdmin = (adminId) =>
     .set('Authorization', cookies.get(CookieTypes.admin.token, {path: '/'}))
     .use(apiPrefix));
 
+
+export const deleteUser = (eventAlias, id) =>
+  promisify(request
```
Comment (RedbackThomson) [inline src/assets/js/data/Api.js]: Elsewhere on the website we only use GET and POST requests. I think a DELETE is fine, but it's the first one to be different.

File: src/assets/js/components/User.js (hunk 1 above)
Comment (subhankar-panda) [reply src/assets/js/components/User.js]: What exactly are you talking about? The logic to show the confirmation buttons?

Comment (RedbackThomson) [reply src/assets/js/components/User.js]: Yeah the render logic, at least. Just to keep things clean

**Comment by RedbackThomson** [comment]
Invalidated by #134 .

**Pull request closed without merging**